nautobot/core/management/commands/build.py

```diff
@@ -0,0 +1,112 @@
+import json
+import os
+import pathlib
+import shlex
+import subprocess
+
+from django.apps import apps
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+import jinja2
+
+
+# TODO: Make this an absolute path relative to where the package is installed.
+NAUTOBOT_UI_DIR = os.path.join(os.path.dirname(settings.BASE_DIR), "nautobot_ui")
+
+
+class Command(BaseCommand):
+
+    help = "Build user interface for Nautobot server environment and installed Nautobot Apps."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--no-npm-install",
+            action="store_false",
+            dest="npm_install",
+            default=True,
+            help="Do not install UI packages.",
+        )
+        parser.add_argument(
+            "--no-render-plugins",
+            action="store_false",
+            dest="render_plugins",
+            default=True,
+            help="Do not render plugin imports.",
+        )
+        parser.add_argument(
+            "--no-npm-build",
+            action="store_false",
+            dest="npm_build",
+            default=True,
+            help="Do not compile UI.",
+        )
+
+    def render_plugin_imports(self):
+        """Render `plugin_imports.js` and update `jsconfig.json` to map to the path for each."""
+        self.stdout.write(self.style.SUCCESS(">>> Rendering plugin imports..."))
+        router_file_path = os.path.join(NAUTOBOT_UI_DIR, "src", "router.js")
+        jsconfig_file_path = os.path.join(NAUTOBOT_UI_DIR, "jsconfig.json")
+        jsconfig_base_file_path = os.path.join(NAUTOBOT_UI_DIR, "jsconfig-base.json")
+
+        with open(jsconfig_base_file_path, "r", encoding="utf-8") as base_config_file:
+            jsconfig = json.load(base_config_file)
+
+        for plugin_path in settings.PLUGINS:
+            plugin_name = plugin_path.split(".")[-1]
+            app_config = apps.get_app_config(plugin_name)
+
+            abs_plugin_path = pathlib.Path(app_config.path).resolve()
+            plugin_path = os.path.relpath(abs_plugin_path, NAUTOBOT_UI_DIR)
+
+            # TODO: Assert that a plugin has a UI folder.
+            jsconfig["compilerOptions"]["paths"][f"@{plugin_name}/*"] = [f"{plugin_path}/ui/*"]
+
+        with open(jsconfig_file_path, "w", encoding="utf-8") as generated_config_file:
+            json.dump(jsconfig, generated_config_file, indent=4)
+
+        plugin_imports_final_file_path = os.path.join(NAUTOBOT_UI_DIR, "src", "plugin_imports.js")
+
+        environment = jinja2.sandbox.SandboxedEnvironment(loader=jinja2.FileSystemLoader(NAUTOBOT_UI_DIR))
```

Comment:
Forgot to add that I made sure this was using sandboxed env.

Reply:
Wonder if there is a Pylint rule for this...